beastling/configuration.py

```diff
@@ -1,56 +1,31 @@
-import collections
-import importlib
-import io
 import itertools
-import os
 import random
 import re
-import sys
-from urllib.request import FancyURLopener
 from pathlib import Path
 from configparser import ConfigParser
 
 import newick
 from appdirs import user_data_dir
 from csvw.dsv import reader
 
-from beastling.fileio.datareaders import load_location_data
-from .distributions import Distribution
-import beastling.clocks.strict as strict
-import beastling.clocks.relaxed as relaxed
+from beastling.fileio.datareaders import iterlocations
 import beastling.clocks.random as random_clock
-import beastling.clocks.prior as prior_clock
 
 import beastling.models.geo as geo
-import beastling.models.binaryctmc as binaryctmc
-import beastling.models.bsvs as bsvs
-import beastling.models.covarion as covarion
-import beastling.models.pseudodollocovarion as pseudodollocovarion
-import beastling.models.mk as mk
 
 from beastling import sections
 from beastling.util import log
+from beastling.util.misc import retrieve_url
 
 import beastling.treepriors.base as treepriors
 from beastling.treepriors.coalescent import CoalescentTree
+from beastling.distributions import Calibration
 
 _BEAST_MAX_LENGTH = 2147483647
 GLOTTOLOG_NODE_LABEL = re.compile(
     "'(?P<name>[^\[]+)\[(?P<glottocode>[a-z0-9]{8})\](\[(?P<isocode>[a-z]{3})\])?(?P<appendix>-l-)?'")
 
 
-class Calibration(
```

Comment:
I turned `Calibration` and `Distribution` into "proper" classes (rather than `namedtuple`s), because I found this multiple inheritance thing a bit hard to understand. In particular the fact that the `from_string` method in the base class `Distribution` was specially crafted for the needs of the derived class `Calibration` (in accepting `**kw`) seemed wrong.